react/app-admin/components/Statistics/SearchTagsChart.tsx

```diff
@@ -3,164 +3,158 @@ import { Form, Select } from 'semantic-ui-react';
-            }
+                key: 'parent_id',
+                value: 'parent_id',
+                text: 'Tags principaux',
+            },
         ];
-        searchTagsStats.map(t => options.push({
-            key: t.slug,
-            value: t.slug,
-            text: t.name,
-        }));
+        searchTagsStats.filter(t => t.is_parent)
+            .map(t =>
+            options.push({
+                key: t.id,
+                value: t.id,
+                text: t.name,
+            }),
+        );
 
         return options;
-    }
+    };
 
     const onTagOptionInputChange = (_: any, { value }: { value: string }): void => {
         setselectedTagOption(value);
         updateChartData(value);
     };
 
-    const updateChartData = (selectedTagOption: string) : void => {
-        
+    const updateChartData = (selectedTagOption: string): void => {
         let labels: string[] = [];
         let data: number[] = [];
-        console.log(searchTagsStats);
-        if(selectedTagOption === "parent_slug"){
+        if (selectedTagOption === 'parent_id') {
             labels = selectDefaultLabels();
             data = selectDefaultData();
         } else {
             labels = selectLabels(selectedTagOption);
             data = selectData(selectedTagOption);
         }
-        console.log()
         updateChart(labels, data);
-    }
+    };
 
-    const selectLabels = (slug: string) : string[] => {
+    const selectLabels = (tag_id: string): string[] => {
         return searchTagsStats
-            .filter(t => t.parent_slug === slug || t.slug === slug)
+            .filter(t => t.parent_id === tag_id || t.id === tag_id)
             .map(t => t.name);
-    }
+    };
 
-    const selectDefaultLabels = () : string[] => {
-        return searchTagsStats
-            .filter(t => t.is_parent)
-            .map(t => t.name);
-    }
+    const selectDefaultLabels = (): string[] => {
+        return searchTagsStats.filter(t => t.is_parent).map(t => t.name);
+    };
 
-    const selectData = (slug: string) : number[] => {
+    const selectData = (tag_id: string): number[] => {
         return searchTagsStats
-            .filter(t => t.parent_slug === slug || t.slug === slug)
+            .filter(t => t.parent_id === tag_id || t.id === tag_id)
             .map(t => t.count);
-    }
+    };
 
-    const selectDefaultData = () : number[] => {
-        return searchTagsStats
-            .filter(t => t.is_parent)
-            .map(t => t.count);
-    }
+    const selectDefaultData = (): number[] => {
+        return searchTagsStats.filter(t => t.is_parent).map(t => t.count);
+    };
 
-    const updateChart = (labels: string[], data: number[]) : void => {
+    const updateChart = (labels: string[], data: number[]): void => {
         while (chart?.data.labels != null && chart.data.labels.length > 0) {
             chart.data.labels.pop();
         }
         labels.map(l => chart?.data.labels?.push(l));
```

Comment:
```suggestion
        labels.forEach(l => chart?.data.labels?.push(l));
```
Same effect but technically `forEach` is what you want here as you don't use the returned result